salesforce/stream-implementer.bal

```diff
@@ -0,0 +1,158 @@
+            self.index += 1;
+            return singleRecord;
+        }
+        return;
+    }
+
+    isolated function fetchQueryResult() returns record {}[]|error {
+        SoqlQueryResult response;
+        if (self.nextRecordsUrl.trim() != EMPTY_STRING) {
+            response = check self.httpClient->get(self.nextRecordsUrl);
+        } else {
+            response = check self.httpClient->get(self.path);
+        }
+        self.nextRecordsUrl = response.hasKey(NEXT_RECORDS_URL) ? check response.get(NEXT_RECORDS_URL).ensureType() :
+            EMPTY_STRING;
+
+        map<json>[] array = response.records;
+        return check covertToRecordsArray(array);
+    }
+}
+
+isolated function covertToRecordsArray(map<json>[] queryResultArray) returns record {}[]|error {
+    record {}[] resultRecordArray = [];
+    foreach map<json> queryResult in queryResultArray {
+        _ = check queryResult.removeIfHasKey(ATTRIBUTES).ensureType();
+        resultRecordArray.push(check queryResult.cloneWithType(Record));
+    }
+    return resultRecordArray;
+}
+
+class SoslSearchResultStream {
+    private record {}[] currentEntries = [];
+    int index = 0;
+    private final http:Client httpClient;
+    private final string path;
+
+    isolated function init(http:Client httpClient, string path) returns error? {
+        self.httpClient = httpClient;
+        self.path = path;
+        self.currentEntries = check self.fetchSearchResult();
+    }
+
+    public isolated function next() returns record {|record {} value;|}|error? {
+        if (self.index < self.currentEntries.length()) {
+            record {|record {} value;|} singleRecord = {value: self.currentEntries[self.index]};
+            self.index += 1;
+            return singleRecord;
+        }
+        return;
+    }
+
+    isolated function fetchSearchResult() returns record {}[]|error {
+        SoslSearchResult response = check self.httpClient->get(self.path);
+        map<json>[] array = response.searchRecords;
+        return check covertSearchResultsToRecordsArray(array);
+    }
+}
+
+isolated function covertSearchResultsToRecordsArray(map<json>[] queryResultArray) returns record {}[]|error {
+    record {}[] resultRecordArray = [];
+    foreach map<json> queryResult in queryResultArray {
+        resultRecordArray.push(check queryResult.cloneWithType(Record));
+    }
+    return resultRecordArray;
+}
+
+type Record record {};
+
+# Define the SOQL result type.
+#
+# + done - Query is completed or not
+# + totalSize - The total number result records
+# + records - Result records
+type SoqlQueryResult record {|
+    boolean done;
+    int totalSize;
+    SoqlRecord[] records;
+    json...;
+|};
+
+# Defines the SOQL query result record type. 
+#
+# + attributes - Attribute record
+type SoqlRecord record {|
+    Attribute attributes?;
+    json...;
+|};
+
+# Defines SOSL query result.
+#
+# + searchRecords - Matching records for the given search string
+type SoslSearchResult record {|
```

Comment:
```suggestion
type SOSLSearchResult record {|
```

Reply:
Fixed in b02c217b06a758477e2c2ae9f981b5960c17658c